### PR TITLE
Backport globalization

### PR DESCRIPTION
**Pull request by Amir-61**
Backporting these 3 PRs of globalization:
https://github.com/strongloop/loopback-datasource-juggler/pull/1022
https://github.com/strongloop/loopback-datasource-juggler/pull/1024
https://github.com/strongloop/loopback-datasource-juggler/pull/1026

This backporting needs very careful review since there were tons of merge conflicts and there is a possibility of doing something wrong!

I would like to get LGTM from @superkhau @0candy @jannyHou individually to make sure everything looks good and merge conflicts have been solved safely.

Thanks a lot!

### CODE REVIEW

**Comment by Amir-61** [comment]
For now I have not squashed in one commit; I can do it if you think it is necessary after review.

File: lib/dao.js [hunk 1]
```diff
@@ -1300,15 +1301,15 @@ DataAccessObject._normalize = function(filter) {
...
+              err = new Error(g.f('The {{order}} %j has invalid direction', token));
               err.statusCode = 400;
               throw err;
             }
           }
           fields.push(token);
         }
       } else {
-        err = new Error(util.format('The order %j is not valid', order[i]));
+        err = new Error('The order %j is not valid', order[i]);
```
Comment (jannyHou) [inline lib/dao.js]: g.f()

File: lib/include.js [hunk 2]
```diff
@@ -217,8 +217,8 @@ Inclusion.include = function(objects, include, options, cb) {
     //}
     if (!relation.modelTo) {
       if (!relation.polymorphic) {
-        cb(new Error('Relation.modelTo is not defined for relation' +
-          relationName + ' and is no polymorphic'));
+        cb(new Error(g.f('{{Relation.modelTo}} is not defined for relation %s and is no polymorphic',
```
Comment (jannyHou) [inline lib/include.js]: {{Polymorphic}}

**Comment by jannyHou** [comment]
LGTM, only two comments 👍

File: package.json [hunk 3]
```diff
@@ -46,6 +46,7 @@
     "loopback-connector": "^2.1.0",
     "node-uuid": "^1.4.2",
     "qs": "^3.1.0",
+    "strong-globalize": "^2.6.0",
```
Comment (superkhau) [inline package.json]: ^2.6.2

File: lib/datasource.js [hunk 4]
```diff
@@ -597,7 +597,7 @@ DataSource.prototype.define = function defineClass(className, properties,
   var args = slice.call(arguments);
 
   if (!className) {
-    throw new Error('Class name required');
+    throw new Error(g.f('Class name required'));
```
Comment (superkhau) [inline lib/datasource.js]: '{{className}} is required'

**Comment by superkhau** [comment]
@Amir-61 Two comments, otherwise LGTM.

**Comment by superkhau** [comment]
@Amir-61 Please squash before merging too.

File: lib/datasource.js [hunk 5]
```diff
@@ -316,16 +317,15 @@ DataSource.prototype.setup = function(name, settings) {
...
+          'it should create `{{connector}}` member of dataSource'));
       }
       this.connected = !err; // Connected now
       if (this.connected) {
         this.emit('connected');
       } else {
         // The connection fails, let's report it and hope it will be recovered in the next call
-        console.error('Connection fails: ', err, '\nIt will be retried for ' +
-          'the next request.');
+        console.error(g.f('Connection fails: %s\nIt will be retried for the next request.', err));
```
Comment (0candy) [inline lib/datasource.js]: just use `g.error` instead of `console.error(g.f`

File: lib/datasource.js (hunk 4 above)
Comment (Amir-61) [inline lib/datasource.js]: @superkhau `Class name` is not a variable like `className `; I don't think if we should skip it? thought?

Comment (Amir-61) [inline lib/datasource.js]: NVM, Actually I agree... I changes it now. Thanks!

**Comment by Amir-61** [comment]
@superkhau @jannyHou @0candy,

Thanks for reviewing! 🙇  I applied your feedback and I'm going to land it after checking CI.

Thanks!

**Pull request merged**